homotopy-model/src/proof.rs

```diff
@@ -391,6 +397,45 @@ impl ProofState {
         Ok(true)
     }
 
+    /// Handler for [Action::Strictify].
+    ///
+    /// Returns an error if generator has dimension zero,
+    /// the generator is not in the signature or the boundaries are not atomic
+    fn strictify(&mut self, generator: Generator) -> Result<bool, ProofError> {
+        let info = self
+            .signature
+            .generator_info(generator)
+            .ok_or(ProofError::UnknownGeneratorSelected)?;
+
+        let Some((source, target)) = info.diagram.atomic_boundaries() else {
+            return Ok(false);
+        };
+        if source.dimension != target.dimension || source == target {
```

Comment:
I'm confused, when can the dimensions ever be different?

Reply:
They can't. I was wrong.